### PR TITLE
CloudWatch: migrate old variable queries with empty array

### LINKED ISSUE #49181: CloudWatch: Accept empty array for variableQuery filters
**What happened**:
The string format variableQueries in pre 8.5 versions supported '[]' as parseable filter values, so we should handle them as well.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by iwysiu**
**What this PR does / why we need it**:
The text variable queries  in versions < 8.5 supported passing in `[]` for no tags, so this pr adds those to the migration.

**Which issue(s) this PR fixes**:

Fixes #49181

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/migrations/variableQueryMigrations.test.ts [hunk 1]
```diff
@@ -35,6 +35,19 @@ describe('variableQueryMigrations', () => {
           expect(query.dimensionFilters).toStrictEqual({});
         });
       });
+      describe('and filter param is empty array', () => {
```
Comment (sarahzinger) [inline public/app/plugins/datasource/cloudwatch/migrations/variableQueryMigrations.test.ts:38]: nitpicky nonblocking suggestion: is there a way to describe `filter param is empty` in a way that is more user-focused so it's a little more clear what this is describing?

File: public/app/plugins/datasource/cloudwatch/migrations/variableQueryMigrations.test.ts [hunk 2]
```diff
@@ -35,6 +35,19 @@ describe('variableQueryMigrations', () => {
...
+        it('should leave an empty filter', () => {
+          const query = migrateVariableQuery(
+            'dimension_values(us-east-1,AWS/RDS,CPUUtilization,DBInstanceIdentifier, [])'
+          );
+          expect(query.queryType).toBe(VariableQueryType.DimensionValues);
+          expect(query.region).toBe('us-east-1');
+          expect(query.namespace).toBe('AWS/RDS');
+          expect(query.metricName).toBe('CPUUtilization');
+          expect(query.dimensionKey).toBe('DBInstanceIdentifier');
+          expect(query.dimensionFilters).toStrictEqual({});
```
Comment (sarahzinger) [inline public/app/plugins/datasource/cloudwatch/migrations/variableQueryMigrations.test.ts:48]: we only need this expect for this test right?

**Review by sarahzinger — APPROVED** [review]
seems reasonable to me!

File: public/app/plugins/datasource/cloudwatch/migrations/variableQueryMigrations.test.ts (hunk 1 above)
Comment (iwysiu) [reply public/app/plugins/datasource/cloudwatch/migrations/variableQueryMigrations.test.ts]: changed it a little, but short of putting `[]` in the test name, i wasn't really sure how to describe it

File: public/app/plugins/datasource/cloudwatch/migrations/variableQueryMigrations.test.ts (hunk 2 above)
Comment (iwysiu) [reply public/app/plugins/datasource/cloudwatch/migrations/variableQueryMigrations.test.ts:43]: removed the extra expects

**Review by aangelisc — APPROVED** [review]
LGTM 😊

**Pull request merged**